### ISSUE TITLE
rsumchisqpow pukes on 0 df

### ISSUE THREAD

**Issue by shabbychef** [shabbychef/sadists#3]
This is broken:

``` r
rsumchisqpow(2,wts=1,df=0,pow=1)
[1] NaN NaN
Warning message:
In rchisq(n, df = dd, ncp = nn) : NAs produced
```

A 0 df chisquare should be identically 0 (or, rather, the ncp).

**Comment by shabbychef** [comment]
It is a little bit more complicated than that. A chi-square with 0 d.f. and non-zero ncp is _not_ identically the ncp. Rather it has variance equal to 4 ncp^2, I believe. A central chi-square with 0 d.f. should be zero, but R's `rchisq` gives a warning and the wrong answer.

**Comment by shabbychef** [comment]
_c.f._ https://bugs.r-project.org/bugzilla/show_bug.cgi?id=16521

**Comment by shabbychef** [comment]
Fixed in R 3.2.3 : http://permalink.gmane.org/gmane.comp.lang.r.announce/168

**Comment by shabbychef** [comment]
Fixed by R 3.2.3, closing.
